web/packages/teleport/src/Assist/Chat/Timestamp.tsx

```diff
@@ -0,0 +1,72 @@
+/*
+Copyright 2023 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import React, { useEffect, useState } from 'react';
+
+import { formatRelative } from 'date-fns';
+
+interface TimestampProps {
+  isoTimestamp: string;
+}
+
+export function Timestamp(props: TimestampProps) {
+  const [date] = useState(() => new Date(props.isoTimestamp));
+  const [, setCounter] = useState(0);
+
+  useEffect(() => {
+    const id = window.setInterval(
+      () => setCounter(count => count + 1),
+      1000 * 60
+    );
+
+    return () => {
+      clearInterval(id);
+    };
+  }, []);
+
+  return <span title={date.toLocaleString()}>{formatDate(date)}</span>;
+}
+
+function formatDate(date: Date) {
```

Comment:
the library we use `date-fns` does this for us, example [here](https://github.com/gravitational/teleport.e/blob/fdf04f5d53b97d72059b93277832fe0100e28b03/web/teleport/src/services/workflow/makeAccessRequest.ts#L83): 

doc: https://date-fns.org/v2.30.0/docs/formatDistanceStrict

Reply:
Yeah, this is still using that library if it can't format it nicely. I added some more date parsing on top of that, so we get nice messages like 'just now', 'an hour ago', instead of it always being "15 seconds ago", etc.